x-pack/plugins/rule_registry/server/plugin.ts

```diff
@@ -86,10 +128,47 @@ export class RuleRegistryPlugin
     });
 
     this.eventLogService = eventLogService;
-    return { ruleDataService, eventLogService };
+
+    return { ruleDataService: this.ruleDataService, eventLogService };
   }
 
-  public start(): RuleRegistryPluginStartContract {}
+  public start(core: CoreStart, plugins: RuleRegistryPluginsStart) {
+    const { logger, alertsClientFactory, security } = this;
+
+    alertsClientFactory.initialize({
+      logger,
+      esClient: core.elasticsearch.client.asInternalUser,
+      // NOTE: Alerts share the authorization client with the alerting plugin
+      getAlertingAuthorization(request: KibanaRequest) {
+        return plugins.alerting.getAlertingAuthorizationWithRequest(request);
+      },
+      securityPluginSetup: security,
+    });
+
+    const getRacClientWithRequest = (request: KibanaRequest) => {
+      return alertsClientFactory.create(request);
+    };
+
+    return {
+      getRacClientWithRequest,
+      alerting: plugins.alerting,
+    };
+  }
+
+  private createRouteHandlerContext = (): IContextProvider<RacRequestHandlerContext, 'rac'> => {
+    const { alertsClientFactory } = this;
+    return async function alertsRouteHandlerContext(
+      context,
+      request
+    ): Promise<RacApiRequestHandlerContext> {
```

Comment:
nit: this function doesn't have to be async

```suggestion
    return function alertsRouteHandlerContext(context, request): RacApiRequestHandlerContext {
```